File: great_escape.py
t = int(input())
for _ in range(t):
    n, k = map(int, input().split())
    total = n * n
    m = total - k
    if m == 1:
        print("NO")
        continue
    print("YES")
    grid = [['R' for _ in range(n)] for _ in range(n)]
    if m == 0:
        pass
    elif n == 2 and m == 3:
        grid[0][0] = 'U'
        grid[0][1] = 'D'
        grid[1][0] = 'R'
        grid[1][1] = 'L'
    else:
        col_trap = []
        col = 0
        rem = m
        while rem >= 2:
            take = min(n, rem)
            col_trap.append((col, take))
            col += 1
            rem -= take
        adjusted = False
        if rem > 0:
            for i in range(len(col_trap) - 1, -1, -1):
                ts = col_trap[i][1]
                if ts >= 3:
                    col_trap[i] = (col_trap[i][0], ts - 1)
                    col_trap.append((col, 2))
                    col += 1
                    adjusted = True
                    break
        for c, size in col_trap:
            for i in range(size - 1):
                grid[i][c] = 'D'
            if size >= 2:
                grid[size - 1][c] = 'U'
    for row in grid:
        print(''.join(row))

File: great_escape.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    
    int t;
    cin >> t;
    for (int test = 0; test < t; ++test) {
        int n, k;
        cin >> n >> k;
        int total = n * n;
        int m = total - k;
        if (m == 1) {
            cout << "NO\n";
            continue;
        }
        cout << "YES\n";
        vector<vector<char>> grid(n, vector<char>(n, 'R'));
        if (m == 0) {
            // already all 'R'
        } else if (n == 2 && m == 3) {
            // special case
            grid[0][0] = 'U';
            grid[0][1] = 'D';
            grid[1][0] = 'R';
            grid[1][1] = 'L';
        } else {
            vector<pair<int, int>> col_trap;
            int col = 0;
            int rem = m;
            while (rem >= 2) {
                int take = min(n, rem);
                col_trap.emplace_back(col, take);
                ++col;
                rem -= take;
            }
            bool adjusted = false;
            if (rem > 0) {
                // rem == 1
                for (int i = (int)col_trap.size() - 1; i >= 0; --i) {
                    int& ts = col_trap[i].second;
                    if (ts >= 3) {
                        --ts;
                        col_trap.emplace_back(col, 2);
                        ++col;
                        adjusted = true;
                        break;
                    }
                }
            }
            // if not adjusted, it would have been handled as special above
            for (auto [c, size] : col_trap) {
                for (int i = 0; i < size - 1; ++i) {
                    grid[i][c] = 'D';
                }
                if (size >= 2) {
                    grid[size - 1][c] = 'U';
                }
            }
        }
        for (int i = 0; i < n; ++i) {
            for (int j = 0; j < n; ++j) {
                cout << grid[i][j];
            }
            cout << '\n';
        }
    }
    return 0;
}